### PR TITLE
Bug-fix: Reorder changes phenotype for individuals with parameters (#261)

### PR DESCRIPTION
**Pull request by HenrikMettler**
Fixes the bug found in #261. Replaces the names in 'parameter_names_to_values' with the updated names (-> addresses). 
The updated test currently fails after some reorderings - I can't understand why, help would be appreciated (I added some lines for debugging to make the error more understandable)

### CODE REVIEW

**Comment by jakobj** [comment]
thanks for working on this @HenrikMettler! why did you remove the leading underscore of `parameter_names_to_values`? this introduces a lot of diffs that are not really relevant changes or am I missing something?

**Comment by HenrikMettler** [comment]
Sorry about that, I changed it during debugging, since assuming it had to be unprotected to replace the values. Reverted in the latest commit.

File: test/test_genome.py [hunk 1]
```diff
@@ -542,10 +542,10 @@ def test_mutate_does_not_reinitialize_parameters(genome_params, rng, mutation_ra
     genome_params["primitives"] = (cgp.Parameter,)
     genome = cgp.Genome(**genome_params)
     genome.randomize(rng)
-    genome._parameter_names_to_values["<p2>"] = math.pi
-    parameter_names_to_values_before = genome._parameter_names_to_values.copy()
+    genome.parameter_names_to_values["<p2>"] = math.pi
```
Comment (jakobj) [inline test/test_genome.py:545]: still missing the preceding underscore

**Review by jakobj — APPROVED** [review]
looks great, thanks! :+1:

could you please squash all commits into one before merging?

**Comment by jakobj** [comment]
excellent :+1: and merging

**Pull request merged**